pylib/cqlshlib/driver.py

```diff
@@ -0,0 +1,145 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import random
+import six
+import stat
+from cassandra.cluster import Cluster
+from cassandra.connection import UnixSocketEndPoint
+from cassandra.policies import WhiteListRoundRobinPolicy, RoundRobinPolicy
+from cassandra.pool import HostDistance
+
+
+class EndpointWhiteListRoundRobinPolicy(WhiteListRoundRobinPolicy):
+    def __init__(self, hosts):
+        self._allowed_hosts = self._allowed_hosts_resolved = tuple(hosts)
+        RoundRobinPolicy.__init__(self)
+
+    def populate(self, cluster, hosts):
+        self._live_hosts = frozenset(h for h in hosts if h.endpoint in self._allowed_hosts_resolved)
+
+        if len(hosts) <= 1:
+            self._position = 0
+        else:
+            self._position = random.randint(0, len(hosts) - 1)
+
+    def distance(self, host):
+        if host.endpoint in self._allowed_hosts_resolved:
+            return HostDistance.LOCAL
+        else:
+            return HostDistance.IGNORED
+
+    def on_up(self, host):
+        if host.endpoint in self._allowed_hosts_resolved:
+            RoundRobinPolicy.on_up(self, host)
+
+    def on_add(self, host):
+        if host.endpoint in self._allowed_hosts_resolved:
+            RoundRobinPolicy.on_add(self, host)
+
+
+def cluster_factory(host, whitelist_lbp=True, **kwargs):
+    """
+    Cluster factory to create a cassandra or dse cluster for cqlsh.
+
+    :param host: the host for the connection. This can be the hostname string or an EndPoint instance.
+    :param whitelist_lbp: Specify if a WhiteListRoundRobinPolicy should be applied. This can be set to True,
+                          False or an EndPoint instance, which is used to configure the policy when
+                          connecting to a cloud cluster. On a cloud cluster, the LBP has to be set after the
+                          nodes discovery. Otherwise, the LBP is set as usual to the execution profile.
+                          Default to True.
+    :kwargs: All other keyword arguments are passed to the Cluster constructor.
+    :return: A Cluster instance.
+    """
+
+    is_cloud_cluster = False
```

Comment:
I am not very happy with the flow in this function.
It should be more like a prose:
- if cloud create cloud
- if not cloud create normal
but I decided to not make it significantly different from original DSE version.